File: services/ai-storage-service/src/services/aiProviders/mock.js
// Zero-dependency provider used by default so the template runs out of the
// box with no API key. Swap AI_PROVIDER in .env once you have real creds.
async function chat({ messages }) {
  const lastUser = [...messages].reverse().find((m) => m.role === 'user');
  return {
    content: `[mock provider] I received: "${(lastUser?.content || '').slice(0, 200)}". Set AI_PROVIDER + AI_API_KEY in .env to use a real model.`,
    usage: { promptTokens: 0, completionTokens: 0 },
  };
}

module.exports = { chat };

File: services/ai-storage-service-python/services/ai_providers/mock.py
from typing import Dict, Any


async def chat(messages: list[Dict[str, str]], model: str = None) -> Dict[str, Any]:
    last_user = next((m for m in reversed(messages) if m["role"] == "user"), None)
    content = f'[mock provider] I received: "{(last_user.get("content", "")[:200])}". Set AI_PROVIDER + AI_API_KEY in .env to use a real model.'
    return {
        "content": content,
        "usage": {"prompt_tokens": 0, "completion_tokens": 0}
    }
